src/conftest.py

```diff
@@ -34,143 +33,156 @@ def _firmware_file_storage_directory() -> str:
+            'file_database': '/tmp/fact_tests_database.log',
+            'level': 'DEBUG',  # Use lowest loglevel for tests
+        },
+        'postgres': {
+            'server': config.common.postgres.server,
+            'port': config.common.postgres.port,
+            'database': config.common.postgres.test_database,
+            'test_database': config.common.postgres.test_database,
+            'ro_user': config.common.postgres.ro_user,
+            'ro_pw': config.common.postgres.ro_pw,
+            'rw_user': config.common.postgres.rw_user,
+            'rw_pw': config.common.postgres.rw_pw,
+            'del_user': config.common.postgres.del_user,
+            'del_pw': config.common.postgres.del_pw,
+            'admin_user': config.common.postgres.admin_user,
+            'admin_pw': config.common.postgres.admin_pw,
         },
-        'unpack': {
-            'base-port': '9900',
-            'max-depth': '10',
-            'memory-limit': '2048',
-            'threads': '2',
-            'whitelist': '',
+        'preset': {
+            'default': {
+                'name': 'default',
+                'plugins': [],
+            },
+            'minimal': {
+                'name': 'minimal',
+                'plugins': [],
+            },
         },
-        'statistics': {'max_elements_per_chart': '10'},
     }
 
-    # Update recursively
-    for section_name in defaults if defaults else {}:
-        sections.setdefault(section_name, {}).update(defaults[section_name])
+    test_config = deep_update(test_config, overwrite_config)
+
+    return config.Common(**test_config)
+
+
+@pytest.fixture
+def backend_config(request, common_config, _firmware_file_storage_directory) -> config.Backend:
+    overwrite_config = merge_markers(request, 'backend_config_overwrite', dict)
+
+    firmware_file_storage_directory = overwrite_config.get(
+        'firmware_file_storage_directory',
+        _firmware_file_storage_directory,
+    )
 
-    configparser_cfg = ConfigParser()
-    configparser_cfg.read_dict(sections)
+    test_config = {
+        'firmware_file_storage_directory': firmware_file_storage_directory,
+        'block_delay': 0.1,
+        'ssdeep_ignore': 1,
+        'intercom_poll_delay': 1.0,
+        'throw_exceptions': True,  # Always throw exceptions to avoid miraculous timeouts in test cases
+        'plugin_defaults': {'processes': 1},
+        'unpacking': {
+            # FIXME the following tests break if we use only one unpacking process
+            # src/test/integration/scheduler/test_cycle_with_tags.py::TestTagPropagation::test_run_analysis_with_tag
+            # src/test/integration/scheduler/test_unpack_and_analyse.py::TestFileAddition::test_unpack_and_analyse
+            # src/test/integration/scheduler/test_unpack_only.py::TestFileAddition::test_unpack_only
+            # src/test/unit/scheduler/test_unpack.py::TestUnpackScheduler::test_unpack_a_container_including_another_container
```

Comment:
the reason for this is that when only one unpacking worker is used, it is not ready when submitting the first file which results in a connection error. The branch [extractor-connection-error-bugfix](https://github.com/fkie-cad/FACT_core/compare/extractor-connection-error-bugfix) adds a retry which fixes this problem but it is probably only relevant for the tests (because no one will probably start FACT with one worker and submit a firmware instantaneously after start).

Reply:
Since #1031 seems to have fixed this, you can probably remove comment